src/frontend/components/app/modal-info.tsx

```diff
@@ -0,0 +1,43 @@
+import { Modal } from '@adminjs/design-system'
+import React from 'react'
+import { useSelector } from 'react-redux'
+import { ModalButton } from '../../interfaces'
+import { ReduxState } from '../../store'
+import { useModal, useTranslation } from '../../hooks'
+
+export const ModalInfo = () => {
+  const rawModal = useSelector((state: ReduxState) => state.modal)
+  const { closeModal } = useModal()
+  const { translateButton, translateLabel, translateMessage } = useTranslation()
+  const modal = { ...rawModal }
+  const close = () => {
+    closeModal()
+  }
+  const confirm = () => {
+    closeModal()
+    modal.confirmAction()
+  }
+  const buttons: Array<ModalButton> = []
+  if (modal.type === 'confirm' && typeof modal.confirmAction === 'function') {
+    buttons.push({ label: translateButton('Cancel', modal.resourceId), variant: 'default', onClick: close })
+    buttons.push({ label: translateButton('Confirm', modal.resourceId), variant: 'primary', onClick: confirm })
+  } else {
+    buttons.push({ label: translateButton('OK', modal.resourceId), variant: 'default', onClick: close })
+  }
+
+  modal.title = modal.title ? translateMessage(modal.title, modal.resourceId) : ''
```

Comment:
Consider using ``modal.${modal.title}`` etc for translation keys so that they are always specific to a modal and not also to general translations with similar names

Reply:
The generic modal translations will be continued in AJS-479